File: ruby/codeeval/008_reverse_words.rb
def reverse(input)
       w = input.chomp.strip.split(" ").to_a
       l = w.length
       (((l - 1) / 2).round).downto(0).each do |i|
             t = w[i]
             w[i] = w[l - i]
             w[l - i] = t
         end
       puts w.join(" ").gsub(/^\s/, '')
       0
end

File.open(ARGV[0]).each_line do |line|
    reverse(line)
end

File: javascript/codeeval/008_reverse_words.js
function reverseWords(input) {
	var words = input.split(' ')
	var res = '';
	for (i = words.length - 1; i > 0; i--) {
		res += words[i] + ' '
	}
	console.log(res + words[0]);
}

var fs  = require("fs");
fs.readFileSync(process.argv[2]).toString().split('\n').forEach(function (line) {
	if (line != "") {
		reverseWords(line);
	}
});